orttraining/orttraining/python/training/optim/_ds_modifier.py

```diff
@@ -10,8 +10,10 @@
 # - has_overflow_partitioned_grads_serial : https://github.com/microsoft/DeepSpeed/blob/d8e9ef6f99e27bb95e10bd146d145b3372b4cfda/deepspeed/runtime/zero/stage2.py#L1799
 # --------------------------------------------------------------------------
 
+import inspect
 import types
 import warnings
+from typing import Dict, List
```

Comment:
You could consider using `from __future__ import annotations` to avoid using typing.* and use the native type directly.